packages/pages/src/components/analytics/analytics.test.tsx

```diff
@@ -0,0 +1,202 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from "react";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { TemplateProps } from "../../common/src/template/types";
+import { AnalyticsProvider } from "./provider";
+import { useAnalytics, useTrack } from "./hooks";
+import { AnalyticsScopeProvider } from "./scope";
+
+global.fetch = jest.fn().mockImplementation(
+  jest.fn(() => {
+    return Promise.resolve({ status: 200 });
+  }) as jest.Mock
+);
+
+jest.spyOn(global, "fetch");
+
+const baseProps: TemplateProps = {
+  document: {
+    __: {
+      entityPageSet: {},
+      name: "foo",
+    },
+    uid: 0,
+    businessId: 0,
+    siteId: 0,
+  },
+  __meta: { mode: "development" },
+};
+
+const currentProcess = global.process;
+
+beforeAll(() => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
```

Comment:
do we need these if we declare them ahead of time?

Reply:
typescript _hates_ mucking with the globals in a manner that's needed for this to work.  I can try to declare but then I get several other typescript errors. I personally don't think it's worth it to try to make typescript happy with this.